File: Two_Pointers/344_Reverse_String/solution344.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        l, r = 0, len(s) - 1
        while l < r:
            s[l], s[r] = s[r], s[l]
            l += 1
            r -= 1

# Runtime1: 172ms, beats 39.07%
# Memory1: 20.89MB, beats 34.11%

# Runtime2: 172ms, beats 39.07%
# Memory2: 20.86MB, beats 34.11%

# Runtime3: 175ms, beats 23.02%
# Memory3: 20.80MB, beats 34.11%

File: Two_Pointers/344_Reverse_String/solution344.cpp
#include <vector>

using namespace std;

class Solution {
public:
    void reverseString(vector<char>& s) {
        int l = 0, r = s.size() - 1;
        while (l < r) {
            swap(s[l], s[r]);
            l++;
            r--;
        }
    }
};

// Time Complexity:
// Space Complexity:

// Runtime1: 20ms, beats 36.80%
// Memory1: 27.12MB, beats 86.58%

// Runtime2: 21ms, beats 25.95%
// Memory2: 27.23MB, beats 66.72%

// Runtime3: 19ms, beats 46.80%
// Memory3: 27.28MB, beats 66.72%